Dict 4 5

AnastasiVokhmyanina28 opened the pull request:
Словарь переписан на Spring

AnastasiVokhmyanina28 committed:
package creation com

создала пакет com
исправила файл .bat

AnastasiVokhmyanina28 committed:
Delete Dictionary.java

AnastasiVokhmyanina28 committed:
:)

добавлен файл pom.xml
 исправлен файл .bat

AnastasiVokhmyanina28 committed:
add

Добавлены библиотеки в pom.xml
добавлен class SpringConfig и Validator
добавлен interface ValidationValue
исправлен classDictionaryType

AnastasiVokhmyanina28 committed:
add

поправлен interface ChoiceOfAction
переписал  class Dictionary
переписал  class FileOperation

AnastasiVokhmyanina28 committed:
edit

редактирование class DictionaryStorage, Console

AnastasiVokhmyanina28 committed:
edit

изменен class Dictionary
изменен class FileOperation
изменен class Console
изменен class DictionaryStorage

AnastasiVokhmyanina28 committed:
edit

исправление properties  файла

AnastasiVokhmyanina28 committed:
edit

изменения в файле pom

AnastasiVokhmyanina28 committed:
edit

AnastasiVokhmyanina28 committed:
resolve conflict

DanilKozlov00 commented on src/main/java/com/Validation/ValidationValue.java:
```diff
@@ -0,0 +1,5 @@
+package com.Validation;
```
Имена пакетов пишутся с маленькой буквы.
И лучше этот пакет поместить в пакет контроллера

DanilKozlov00 commented on src/main/java/com/config/SpringConfig.java:
```diff
@@ -0,0 +1,72 @@
+    public Map<Integer, DictionaryType> getDictionaryTypeMap() {
+        Map<Integer, DictionaryType> dictionaryMap = new HashMap<>();
+        for (Map.Entry<Integer, String> pair : dictionaryTypeMap.entrySet()) {
+            String[] m = pair.getValue().split(",");
```
Имена переменных нужно делать осознанными, что бы понимать что они хранят.
Разделитель вынести в поле. И организовать получение из файла properties. Вдруг создание файла словарей будет производится в Excel и сохраниться с разделителем ';'. Тогда придется лезть в код и менять строку. А так изменим только файл с настройками.

DanilKozlov00 commented on src/main/java/com/config/SpringConfig.java:
```diff
@@ -0,0 +1,72 @@
+        return dictionaryMap;
+    }
+
+    @Bean("ds")
```
Непонятное имя бина. Сделать более осознанное.

DanilKozlov00 commented on src/main/java/com/config/SpringConfig.java:
```diff
@@ -0,0 +1,72 @@
+
+    @Bean("file")
+    public DictionaryType getFileDictionary() {
+        return getDictionaryTypeMap().get(2);
```
Не понятно что за 1 и 2. вынести в константы с именем отображающим их назначение,

DanilKozlov00 commented on src/main/java/com/config/SpringConfig.java:
```diff
@@ -0,0 +1,72 @@
+        return new Console();
+    }
+
+    @Bean
+    public Dict getDict() {
+        return new Dict();
+    }
```
Бесполезный бин, нигде не используется.

DanilKozlov00 commented on src/main/java/com/Dict.java:
```diff
@@ -0,0 +1,14 @@
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class Dict {
```
Раз это точка входа в программу, то разумнее переименовать в Main класс

DanilKozlov00 commented on src/main/java/com/config/SpringConfig.java:
```diff
@@ -0,0 +1,72 @@
+@PropertySource("classpath:config.properties")
+public class SpringConfig {
+    @Value("#{${valuesMap}}")
+    Map<Integer, String> dictionaryTypeMap;
```
Модификатор доступа где?

DanilKozlov00 commented on src/main/java/com/controller/FileOperation.java:
```diff
@@ -1,38 +1,43 @@
+        this.path = dictionaryType.getDictionaryPath();
     }
 
     @Override
     public String fileReading() {
         StringBuilder data = new StringBuilder();
         try {
-            File file = new File(path);
+            File file = new ClassPathResource(path).getFile();
             Scanner scanner = new Scanner(file);
             while (scanner.hasNextLine()) {
                 data.append(scanner.nextLine()).append("\n");
             }
-        } catch (FileNotFoundException e) {
+        } catch (IOException e) {
```
Сканер не закрыт, после чтения. 
Либо закрываем сканер в блоке finally, либо используем try with resources.
Желательно, использовать специально созданные BufferReader для чтения файла, а не сканеры.

DanilKozlov00 commented on src/main/java/com/controller/FileOperation.java:
```diff
@@ -1,38 +1,43 @@
     @Override
     public String addAnEntry(String key, String value) {
-        if (keyCheck(key) && valueCheck(value)) {
+        if (validator.get(dictionaryType).validPair(key, value)) {
```
Лучше тоже переделать на try-with-resources

DanilKozlov00 commented on src/main/java/com/view/Console.java:
```diff
@@ -46,40 +47,29 @@ public int startp() {
     }
 
-    private void startS(int choice) {
+    private void startS() {
```
Непонятное имя метода, дать осознанное, отражающее суть его работы.

DanilKozlov00 commented on src/main/java/com/view/Console.java:
```diff
@@ -46,40 +47,29 @@ public int startp() {
-    public static FileOperation creation1(DictionaryType dictionaryType) {
-        String dictionaryPath = dictionaryType.getDictionaryPath();
-        return new FileOperation(dictionaryPath, dictionaryType);
+        var data = new DictionaryStorage(dictionaryPath).getData();
```
Переменная нигде не используется, да и метод в целом может быть убран.
Так как просто создает новый объект, что делается в одну строку в методе выше.
new Dictionary(pair.getValue())

DanilKozlov00 commented on src/main/java/com/view/Console.java:
```diff
@@ -46,40 +47,29 @@ public int startp() {
-           this.startS(choice);
-       }
+         while (mapDictionaries == null) {
+             this.startp();
+             this.startS();
```
По именам методов непонятно что они делают, да и this здесь не нужен

DanilKozlov00 commented on src/main/java/com/config/DictionaryType.java:
```diff
@@ -1,17 +1,14 @@
-package config;
-public enum DictionaryType {
+package com.config;
```
Явно не тот пакет для этого класса

DanilKozlov00 reviewed (commented):
Есть над чем подумОть!

AnastasiVokhmyanina28 committed:
fix bug

corrected the package name.
transfer of package validation to package controller

AnastasiVokhmyanina28 committed:
fix bug

deleted bean's name

AnastasiVokhmyanina28 committed:
fix bug

AnastasiVokhmyanina28 committed:
deleted the bean.

AnastasiVokhmyanina28 committed:
fix bug

renamed the Dict class to the Main class

AnastasiVokhmyanina28 committed:
fix bug

Added access modifier

AnastasiVokhmyanina28 committed:
fix bug

AnastasiVokhmyanina28 committed:
fix bug

AnastasiVokhmyanina28 committed:
class editing SpringConfig, FileOperation, DictionaryStorage

DanilKozlov00 reviewed (commented):
Reviewed

AnastasiVokhmyanina28 committed:
fix bug

AnastasiVokhmyanina28 committed:
fix bug

DanilKozlov00 commented on src/main/java/com/model/DictionaryStorage.java:
```diff
@@ -20,31 +15,35 @@ public DictionaryStorage(String path) {
     }
 
-    private  Map<String, String>  parseLine (String line) {
+    private Map<String, String> parseLine(String line) {
```
Метод возвращает map'у при этом она нигде не используется.
Наверное, лучше сделать его void

DanilKozlov00 commented on src/main/java/com/model/DictionaryStorage.java:
```diff
@@ -20,31 +15,35 @@ public DictionaryStorage(String path) {
-        File file = new File(path);
         BufferedWriter bf = null;
         try {
+            File file = new ClassPathResource(path).getFile();
```
Можно переделать на try with resources b избавиться от блока finally и обработки ошибок в нем

DanilKozlov00 commented on src/main/java/com/model/DictionaryStorage.java:
```diff
@@ -20,31 +15,35 @@ public DictionaryStorage(String path) {
         try {
             File file = new ClassPathResource(path).getFile();
-            Scanner scanner = new Scanner(file);
+            scanner = new Scanner(file);
```
Тут не понятно почему сканнер, если во всех других методах чтение идет через BufferReader. 
Можно переделать как в методе fileReading в классе FileOperation.
И так же избавиться от блока finally

DanilKozlov00 reviewed (commented):
Reviewed

AnastasiVokhmyanina28 committed:
fix bug

AnastasiVokhmyanina28 committed:
edit

removed the save method call
